# This file is part of HiGFXback

# requires
REQUIRES="directfb2-build egl-opengl-stubs-build meson-build"

PKG_CONFIG_PATH=/dfb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

pkg-config --exists check-build libfiu-build && REQUIRES="$REQUIRES check-build libfiu-build"

# configure
PKG_CONFIG_PATH=/dfb/lib/pkgconfig meson setup -Dfbdev=false --prefix=/dfb build

# build
meson compile -C build

# install
install -d $DESTDIR/dfb/include/GL
install -m 644 glut.h $DESTDIR/dfb/include/GL/tinyglut.h
ln -sf tinyglut.h $DESTDIR/dfb/include/GL/glut.h
install -d $DESTDIR/dfb/lib
install build/libglut.so.3.*[!p] $DESTDIR/dfb/lib/libtinyglut.so.3
ln -sf libtinyglut.so.3 $DESTDIR/dfb/lib/libglut.so
ln -sf libtinyglut.so.3 $DESTDIR/dfb/lib/libglut.so.3
install -d $DESTDIR/dfb/lib/glut/backends
install build/dfbgl_plugin.so $DESTDIR/dfb/lib/glut/backends
install build/egl_plugin.so $DESTDIR/dfb/lib/glut/backends
install -d $DESTDIR/dfb/lib/glut/platforms
install build/directfb_plugin.so $DESTDIR/dfb/lib/glut/platforms
install -d $DESTDIR/dfb/lib/pkgconfig
install -m 644 build/meson-private/glut.pc $DESTDIR/dfb/lib/pkgconfig/tinyglut.pc
ln -sf tinyglut.pc $DESTDIR/dfb/lib/pkgconfig/glut.pc

# build.pc
install -d $DESTDIR/dfb/share/pkgconfig
cat > $DESTDIR/dfb/share/pkgconfig/tinyglut-build.pc << EOF
Name: TinyGLUT
Version: 0.6
Description: OpenGL Utility Toolkit
Requires: $REQUIRES

devel=\\
/dfb/include/GL/glut.h \\
/dfb/include/GL/tinyglut.h \\
/dfb/lib/libglut.so \\
/dfb/lib/pkgconfig/glut.pc \\
/dfb/lib/pkgconfig/tinyglut.pc

exec=\\
/dfb/lib/libglut.so.3 \\
/dfb/lib/glut/backends/dfbgl_plugin.so \\
/dfb/lib/glut/backends/egl_plugin.so \\
/dfb/lib/glut/platforms/directfb_plugin.so \\
/dfb/lib/libtinyglut.so.3
EOF
ln -sf tinyglut-build.pc $DESTDIR/dfb/share/pkgconfig/glut-build.pc

exit

# patch
--- TinyGLUT-0.6.orig/CMakeLists.txt
+++ TinyGLUT-0.6/CMakeLists.txt
@@ -82,7 +82,7 @@
   endif()
 endif()
 
-if(NOT ENABLE_DUMMY AND NOT ENABLE_X11 AND NOT ENABLE_X11 AND NOT ENABLE_FBDEV AND NOT ENABLE_WAYLAND)
+if(NOT ENABLE_DUMMY AND NOT ENABLE_X11 AND NOT ENABLE_DIRECTFB AND NOT ENABLE_FBDEV AND NOT ENABLE_WAYLAND)
   message(FATAL_ERROR "No platforms found")
 endif()
 
--- TinyGLUT-0.6.orig/meson.build
+++ TinyGLUT-0.6/meson.build
@@ -177,6 +177,8 @@
   else
     add_global_arguments('-DFIU_ENABLE', language: 'c')
   endif
+else
+  libfiu_dep = disabler()
 endif
 
 #######################

# source
https://github.com/caramelli/TinyGLUT/archive/v0.6/TinyGLUT-0.6.tar.gz
